tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -464,6 +464,99 @@ class SerializableModelLibraryGenerator {
     });
   }
 
+  Expression _buildClonableCopyWithAssignment(
+      SerializableModelFieldDefinition field) {
+    var shouldDeepClone = field.type.isListType || field.type.isMapType;
+
+    if (!shouldDeepClone) {
+      return refer(cloneTrivialFunctionName, serverpodUrl(serverCode))
+          .call([refer(Keyword.thisKeyword).property(field.name)]);
+    }
+
+    return _buildDeepCloneTree(
+      field.type,
+      field.name,
+      root: true,
+    );
+  }
+
+  Expression _buildDeepCloneTree(TypeDefinition type, String variableName,
+      {int depth = 0, bool root = false}) {
+    var isLeafNode = type.generics.isEmpty;
+    if (isLeafNode) {
+      return refer(cloneTrivialFunctionName, serverpodUrl(serverCode))
+          .call([refer(variableName)]);
+    }
+
+    var nextCallback = switch (type.className) {
+      ListKeyword.className =>
+        _buildListCloneCallback(type.generics.first, depth),
+      MapKeyword.className =>
+        _buildMapCloneCallback(type.generics[0], type.generics[1], depth),
+      _ => throw "Can't clone type ${type.className}",
+    };
+
+    Expression expression = switch (root) {
+      true => refer(Keyword.thisKeyword).property(variableName),
+      false => refer(variableName),
+    };
+
+    expression = switch (type.nullable) {
+      true => expression.nullSafeProperty(Keyword.mapFunctionName),
+      false => expression.property(Keyword.mapFunctionName),
+    }
+        .call([nextCallback]);
+
+    return type.isListType
+        ? expression.property(ListKeyword.toList).call([])
+        : expression;
+  }
+
+  Expression _buildListCloneCallback(TypeDefinition type, int depth) {
+    var variableName = 'e$depth';
```

Comment:
Comment: Neat addition with the depth recorded in the variable name. Maybe we should add this to the `fromJson` structure as well? I don't think we record the depth there.

Reply:
Makes total sense! Yea, a follow-up PR would be good, then we maintain the same style across the generated code.